test/parallel/test-timers-clearImmediate.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+require('../common');
+const assert = require('assert');
+
+const N = 3;
+var count = 0;
+function next() {
+  const immediate = setImmediate(function() {
+    clearImmediate(immediate);
+    ++count;
+  });
+}
+for (var i = 0; i < N; ++i)
+  next();
+
+process.on('exit', () => {
+  assert.strictEqual(count, N, `Expected ${N} immediate callback executions`);
```

Comment:
Nit: Maybe even go a bit further?

```js
'use strict';
const common = require('../common');

function next() {
  const immediate = setImmediate(common.mustCall(function callback() {
    clearImmediate(immediate);
  }));
}
for (var i = 0; i < 3; ++i)
  next();
```

One benefit is that we don't lose the information of what `count` was if the assertion fires.